### ISSUE TITLE
[Feature Request] support running multiple elements concurrently

### ISSUE THREAD

**Issue by mjsambol** [zacs/ha-dualmodegeneric#28]
Some HVAC systems require that circuits be controlled independently for fan, compressor, heat pump, etc. The fan may be turned on and off together with the compressor, or the fan may run continuously, independently of the compressor's activation and deactivation. There are merits to each of these, see for example

- [reference 1](https://www.rhoadsenergy.com/should-i-set-my-hvac-systems-fan-setting-to-auto-or-on/)
- [reference 2](https://www.lancasterbrothers.com/lancaster-brothers-heating-cooling-blog/how-to-set-ac-fan/)
- [reference 3](https://trusthomesense.com/blog/pros-cons-continuously-running-hvac-fan/)

I wished to have both options, and use the "cool" mode for continuous fan / periodic compressor, and "dryer" mode for synchronized fan + compressor. This matches the behavior of a thermostat I used previously.

Using the ha-dualmodegeneric component to implement fan+compressor together was straightforward, but I had difficulty implementing continuous fan + periodic compressor. The reason is that the code in `_async_control_heating` prevents mode activation while a different mode is active.

I was able to work around this (see [issue 27](https://github.com/zacs/ha-dualmodegeneric/issues/27)) by obscuring the fan controller behind a template switch. For the benefit of future users, I suggest either mentioning this in the documentation or offering a configuration option which would change the behavior of `_async_control_heating` so that the fan being in operation would not prevent other modes from being activated concurrently.

**Comment by zacs** [comment]
I agree this sort of behavior is ideal in HVAC systems, but it seems like a great deal of complexity. Would it work for you to solve each situation like so?

- continuous fan / periodic compressor: I believe this will work in the component as-is, as long as you specify the fan switch and keep it turned on, and the compressor as the cooler switch which would only turn on when needed to hit a target temperature. 
- dryer mode with continuous fan / continuous compressor: Create a switch template that turns on both fan and compressor, and assign that switch template to be the `dryer`. 

That may cause some problems if you turn on the dryer but then manually turn off the fan, etc, but it seems like the simplest way to do what you want.

**Comment by mjsambol** [comment]
Yes, I've solved the issue for myself, but it's a bit involved. I opened the FR for the benefit of future users:

> For the benefit of future users, I suggest either mentioning this in the documentation or offering a configuration option which would change the behavior of _async_control_heating so that the fan being in operation would not prevent other modes from being activated concurrently.
